text/0000-decouple-dependencies.md

```diff
@@ -0,0 +1,403 @@
+```toml
+[[versions]]
+cpes = [ "cpe:2.3:a:apache:maven:3.8.6:*:*:*:*:*:*:*" ]
+name = "Apache Maven"
+purl = "pkg:generic/apache-maven@3.8.6"
+checksum = "sha256:c7047a48deb626abf26f71ab3643d296db9b1e67f1faa7d988637deac876b5a9"
+arch = "x86_64"
+os = "linux"
+distro = "ubuntu-18.04"
+uri = "https://repo1.maven.org/maven2/org/apache/maven/apache-maven/3.8.6/apache-maven-3.8.6-bin.tar.gz"
+version = "3.8.6"
+strip-components = 1
+
+  [[versions.licenses]]
+  type = "Apache-2.0"
+  uri = "https://www.apache.org/licenses/"
+
+[[versions]]
+cpes = [ "cpe:2.3:a:apache:mvnd:0.7.1:*:*:*:*:*:*:*" ]
+name = "Apache Maven Daemon"
+purl = "pkg:generic/apache-mvnd@0.7.1"
+chekcsum = "sha256:ac0b276d4d7472d042ddaf3ad46170e5fcb9350981af91af6c5c13e602a07393"
+arch = "x86_64"
+os = "linux"
+uri = "https://github.com/apache/maven-mvnd/releases/download/0.7.1/mvnd-0.7.1-linux-amd64.zip"
+version = "0.7.1"
+
+  [[versions.licenses]]
+  type = "Apache-2.0"
+  uri = "https://www.apache.org/licenses/"
+```
+## Dependency Validation
+In `buildpack.toml` we will add a section to metadata for specifying dependency
+validation parameters. This is a way that buildpacks can state that they do or
+do not support certain versions of dependencies.
+
+A buildpack does not need to include this section, it is optional. If included,
+the buildpack and libraries like `libpak` and `packit` may use the information
+to fail if dependency versions are requested by a user that might cause
+problems for the buildpack. It is the buildpacks responsibility to process the
+validations and react to them, whether that be warning the user or even
+failing.
+
+The format for this metadata is such that you have an array of tables called
+`validations`. Each table in the array contains the dependency id, which
+follows the format of a dependency id as outlined in the [Metadata
+Format](#metadata-format) section. In addition, it includes an array of strings
+called `supported` which contains a list of [semver](https://semver.org/)
+ranges that indicate what is supported by that buildpack. Optionally it can
+contain a `type` which defaults to `semver`, but can be set to `regex`.
+
+It is recommended that buildpack authors use semver as the type because
+matching is generally simpler that way, however, if a dependency does not
+follow semver you may use regular expressions to match the versions that are
+compatible with the buildpack.
+
+If any semver range or regular expression matches then it can be assumed that
+the buildpack is compatible. If no range matches then it can be assumed that
+the version is not compatible.
+
+For example:
+```toml
+[[metadata.validations]]
+dependency-id = "com.example.jre"
+supported = [ "8.0.*", "11.0.*", "17.0.*" ]
+
+[[metadata.validations]]
+dependency-id = "com.example.nodejs"
+supported = [ "^16.0", "^17.0", "^18.0" ]
+
+[[metadata.validations]]
+dependency-id = "com.example.tomcat"
+supported = [ "8\.5\.\d+", "9\.0\.\d+", "10\.0\.\d+" ]
+type = “regex”
```

Comment:
I think you have special quotes around `regex` - possibly copied from google docs?

Reply:
cc @ForestEckhardt